### PR TITLE
add fundamentals 2: components

### PR DESCRIPTION
**Pull request by luke-m**

### CODE REVIEW

File: src/04-Fundamentals-2.mdx [hunk 1]
```diff
@@ -1,4 +1,121 @@
...
+### Step 1: Install Node.js and NPM
+
+Node.js is a way to run Javascript code without the need to use a browser. It's mainly used to run servers. You won't need to use it a lot, so it's not mandatory to do any work with it for this chapter.  
+Node.js comes with NPM, its very own package manager. This tool is used by the majority of web developers to install packages, libraries and other tools. It's pretty much your bread and butter and it is very much obligatory that you learn how to use it. The first step is to download and install it from [the Node.js homepage](https://nodejs.org/en/).  
+To ensure the installation worked correctly, type the following command into your command line:  `node -v && npm -v`  
+If you don't get any errors, everything worked out as planned. Congrats!
+
+### Step 2: Install Yarn
+
+Yarn is essentially the same as NPM, just faster and a bit better in general. It completes the same tasks as NPM though, that's all you need to know for now. Go ahead and get it from [the Yarn homepage](https://yarnpkg.com/lang/en/docs/install/#windows-stable).
```
Comment (jurekbarth) [inline src/04-Fundamentals-2.mdx]: I would remove Yarn here. Yarn's advantages are super low and we don't need any of it's features. If you still want to keep it i would go for: Yarn is an alternative to NPM we use here at VI...

File: src/04-Fundamentals-2.mdx [hunk 2]
```diff
@@ -1,4 +1,121 @@
...
+If you don't get any errors, everything worked out as planned. Congrats!
+
+### Step 2: Install Yarn
+
+Yarn is essentially the same as NPM, just faster and a bit better in general. It completes the same tasks as NPM though, that's all you need to know for now. Go ahead and get it from [the Yarn homepage](https://yarnpkg.com/lang/en/docs/install/#windows-stable).  
+Try `yarn -v` to check if the installation worked.
+
+### Step 3: Install gulp
+
+gulp is a taskrunner, which does a lot of work for you like compile, build, serve and other stuff. To install it, run the command `npm install gulp-cli -g` and then `npm install gulp -D`.
```
Comment (jurekbarth) [inline src/04-Fundamentals-2.mdx]: 1. Why do we need the gulp-cli? I would prefer to not install anything global, thats not needed global.
2. Gulp itself does not do a lot. The plugins to the heavy lifting. From their website: "gulp is a toolkit for automating painful or time-consuming tasks in your development workflow, so you can stop messing around and build something." 
3. Switching to npm here again. Consider to stay with yarn or npm 🙃

File: src/04-Fundamentals-2.mdx [hunk 3]
```diff
@@ -1,4 +1,121 @@
...
+
+Yarn is essentially the same as NPM, just faster and a bit better in general. It completes the same tasks as NPM though, that's all you need to know for now. Go ahead and get it from [the Yarn homepage](https://yarnpkg.com/lang/en/docs/install/#windows-stable).  
+Try `yarn -v` to check if the installation worked.
+
+### Step 3: Install gulp
+
+gulp is a taskrunner, which does a lot of work for you like compile, build, serve and other stuff. To install it, run the command `npm install gulp-cli -g` and then `npm install gulp -D`. 
+> *That's how easy it is to install packages with NPM, how neat!*  
+
+Gulp works with a so called `gulpfile.js`, which defines which tasks gulp completes for you. `gulpfiles` are written in Node.js (so essentially JS), for this task, we will provide you with one.
```
Comment (jurekbarth) [inline src/04-Fundamentals-2.mdx]: gulpfiles are written in javascript, so is Gulp. To execute that javascript on your computer we use NodeJS as a javascript runtime. NodeJS is using the same as the chrome browser btw.

File: src/04-Fundamentals-2.mdx [hunk 4]
```diff
@@ -1,4 +1,121 @@
...
+
+### Step 3: Clone the framework
+
+Now that you have all the tools you need, it's time to get your hands dirty and start working on the framework. You already know how to use Git to your advantage, so go ahead and clone [the GitHub repository](https://github.com/virtualidentityag/almost-perfect-framework).
```
Comment (jurekbarth) [inline src/04-Fundamentals-2.mdx]: Would rename repo for visitors it could look like our build framework :)

File: src/04-Fundamentals-2.mdx [hunk 5]
```diff
@@ -1,4 +1,121 @@
...
++ Adhere to the BEM naming structure closely as it is absolutely vital to learn it in order to write good biotope components later on
++ Use the BEM way to make modifiers for the component (see reference)
+
+### Resources
++ [SASS's own reference](https://sass-lang.com/guide) or [CSS-Tricks](https://css-tricks.com/snippets/sass/)
++ [BEM's own reference](http://getbem.com/naming/) or [CSS-Tricks](https://css-tricks.com/bem-101/)
+
+## JavaScript
+
+'Finally', is what you're thinking to yourself probably. Yes, now we're getting into the good stuff. Up until now, everything you have coded is more or less 'static'. Well, that changes now. In this section, you will get to Javascript, it's newest version ECMAScript2015 (also called ES6) and, to a lesser extent, jQuery. We here at VI try to write our code in ES6 whenever possible. jQuery, while it used to be the standard and provides a lot of functionality that makes our lives easier, is being used less and less and it's more than likely that it will completely die eventually. That is why coding in 'plain' JS is always preferred.
```
Comment (jurekbarth) [inline src/04-Fundamentals-2.mdx]: We try to write Typescript if possible, so maybe we should add that here somewhere i think 👍

File: src/04-Fundamentals-2.mdx [hunk 6]
```diff
@@ -1,4 +1,121 @@
...
+
+Copy the component you built in the previous step. When one of the buttons is clicked and the corresponding textbox is opened, the other textbox has to close. The rule is: Only one call-to-action textbox can be opened at all times.  
+
+### Hints and guidelines
+
++ As stated before, JS/ES6 is preffered. However, jQuery provides a very cool way of implementing events and adds functionality to them, that doesn't exist in plain JS. Try completing the challenge twice, once in plain JS and once in jQuery to see the difference
++ Optional challenge: Perhaps you are familiar with a more modern approach to component communication: State management. If you want to, you can complete the challenge using Redux or something similiar as well 
+
+### Resources
+
```
Comment (jurekbarth) [inline src/04-Fundamentals-2.mdx]: https://medium.freecodecamp.org/learn-these-javascript-fundamentals-and-become-a-better-developer-2a031a0dc9cf
http://2ality.com/2015/08/getting-started-es6.html

File: src/04-Fundamentals-2.mdx (hunk 1 above)
Comment (luke-m) [reply src/04-Fundamentals-2.mdx]: I don't think we're gonna remove Yarn from all the old projects anytime soon, so I'll leave it in here. I changed the wording a bit, according your suggestion.

File: src/04-Fundamentals-2.mdx (hunk 2 above)
Comment (luke-m) [reply src/04-Fundamentals-2.mdx]: I just pulled this from the gulp website. Removing the gulp-cli.

File: src/04-Fundamentals-2.mdx (hunk 5 above)
Comment (luke-m) [reply src/04-Fundamentals-2.mdx]: Typescript is part of a later chapter. I am leaving it out here.

**Comment by clemensf** [comment]
For consistency: please add separators between sections / challenges (like in chapter fundamentals 1)
\<br/\>

\---------------------------------------

**Pull request merged**